.teamcity/_self/projects/WebApp.kt

```diff
@@ -874,18 +874,124 @@ object HelpCentreE2ETests : E2EBuildType(
 	}
 )
 
-object KPIDashboardTests : E2EBuildType(
-	buildId = "Calypso_E2E_KPI_Dashboard",
-	buildUuid = "441efac5-721a-4557-9448-9234e89fb6b1",
-	buildName = "Test build for KPI Dashboard project",
-	buildDescription = "Test build configuration for KPI dashboard.",
-	testGroup = "kpi",
-	buildParams = {
+object KPIDashboardTests : BuildType({
+	id("Calypso_E2E_KPI_Dashboard")
+	uuid = "441efac5-721a-4557-9448-9234e89fb6b1"
+	name = "Test build for KPI Dashboard project"
+	description = "Test build configuration for KPI dashboard."
+	artifactRules = """
+		logs.tgz => logs.tgz
+		screenshots => screenshots
+		trace => trace
+		allure-results.tgz => allure-results.tgz
+	""".trimIndent()
+
+	vcs {
+		root(Settings.WpCalypso)
+		cleanCheckout = true
+	}
+
+	params {
+		param("env.NODE_CONFIG_ENV", "test")
+		param("env.PLAYWRIGHT_BROWSERS_PATH", "0")
+		param("env.TEAMCITY_VERSION", "2021")
+		param("env.HEADLESS", "false")
+		param("env.LOCALE", "en")
+		param("env.DEBUG", "")
 		param("env.VIEWPORT_NAME", "desktop")
-	},
-	buildFeatures = {
-	},
-)
+		param("env.ALLURE_RESULTS_PATH", "allure-results")
+	}
+
+	steps {
+		bashNodeScript {
+			name = "Prepare environment"
+			scriptContent = """
+				# Install deps
+				yarn workspaces focus wp-e2e-tests @automattic/calypso-e2e
+
+				# Decrypt secrets
+				# Must do before build so the secrets are in the dist output
+				E2E_SECRETS_KEY="%E2E_SECRETS_ENCRYPTION_KEY_CURRENT%" yarn workspace @automattic/calypso-e2e decrypt-secrets
+
+				# Build packages
+				yarn workspace @automattic/calypso-e2e build
+			""".trimIndent()
+			dockerImage = "%docker_image_e2e%"
+		}
+
+		bashNodeScript {
+			name = "Run tests"
+			scriptContent = """
+				# Configure bash shell.
+				shopt -s globstar
+				set -x
+
+				# Enter testing directory.
+				cd test/e2e
+				mkdir temp
+
+				# Run suite.
+				xvfb-run yarn jest --reporters=jest-teamcity --reporters=default --maxWorkers=%E2E_WORKERS% --group=kpi
+			"""
+			dockerImage = "%docker_image_e2e%"
+		}
+
+		bashNodeScript {
+			name = "Collect results"
+			executionMode = BuildStep.ExecutionMode.RUN_ON_FAILURE
+			scriptContent = """
+				set -x
+
+				mkdir -p screenshots
+				find test/e2e/results -type f \( -iname \*.webm -o -iname \*.png \) -print0 | xargs -r -0 mv -t screenshots
+
+				mkdir -p logs
+				find test/e2e/results -name '*.log' -print0 | xargs -r -0 tar cvfz logs.tgz
+
+				mkdir -p trace
+				find test/e2e/results -name '*.zip' -print0 | xargs -r -0 mv -t trace
+
+				mkdir -p allure-results
+				find test/e2e/allure-results -name '*.json' -print0 | xargs -r -0 tar cvfz allure-results.tgz
+			""".trimIndent()
+			dockerImage = "%docker_image_e2e%"
+		}
+	}
+
+	features {
+		perfmon {
+		}
+		commitStatusPublisher {
```

Comment:
**Suggestion:** We can remove this right? We don't need to send this info back to GitHub in any way, do we?

Reply:
Done - removed the commit publisher.